rectools/models/sasrec.py

```diff
@@ -753,86 +517,55 @@ def get_device(self) -> torch.device:
         return self.item_emb.weight.device
 
     @classmethod
-    def from_config(cls, cfg: ItemModelConfig, dataset_stats: DatasetStats):
+    def from_config(cls, hidden_units_item: int, dataset_stats: DatasetStats):
         return cls(
-            embedding_dim=cfg.hidden_units,
+            embedding_dim=hidden_units_item,
             item_num=dataset_stats.item_num,
         )
 
 
-class SASRecConfig(BaseConfig):
-    num_blocks: int
-    hidden_units: int
-    num_heads: int
-    maxlen: int
-    dropout_rate: float
-    item_model: ItemModelConfig
-    use_pos_emb: bool = True
-    use_sm_head: bool = False
-
-
-class PointWiseFeedForward(torch.nn.Module):
-    def __init__(self, hidden_units, dropout_rate):
-        super(PointWiseFeedForward, self).__init__()
-
-        self.conv1 = torch.nn.Conv1d(hidden_units, hidden_units, kernel_size=1)
-        self.dropout1 = torch.nn.Dropout(p=dropout_rate)
-        self.relu = torch.nn.ReLU()
-        self.conv2 = torch.nn.Conv1d(hidden_units, hidden_units, kernel_size=1)
-        self.dropout2 = torch.nn.Dropout(p=dropout_rate)
-
-    def forward(self, inputs):
-        outputs = self.dropout2(self.conv2(self.relu(self.dropout1(self.conv1(inputs.transpose(-1, -2))))))
-        outputs = outputs.transpose(-1, -2)  # as Conv1D requires (N, C, Length)
-        outputs += inputs
-        return outputs
-
-
-def xavier_normal_init(model: nn.Module) -> None:
-    for name, param in model.named_parameters():
-        try:
-            torch.nn.init.xavier_normal_(param.data)
-        except Exception as err:
-            logger.info("undable to init param %s with xavier: %s", name, err)
-            pass  # just ignore those failed init layers
-
+# TODO make list of all models and list it during error
 
-def to_device(x, device: torch.device):
-    if isinstance(x, torch.Tensor):
-        return x.to(device)
 
-    if not isinstance(x, tuple) and not isinstance(x, list):
-        raise Exception(f"expected Tuple, List or Tensor, found {type(x)}")
+def build_item_model(name: str, hidden_units_item: int, dataset_stats: DatasetStats):
+    if name == "idemb":
```

Comment:
We support only one name. So we don't need the method to get item model. And we don't need the parameter ''name"